File: backend/app/models/campaign_active.py
from __future__ import annotations

import uuid
from datetime import datetime

from sqlalchemy import DateTime, String, func, text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class CampaignActive(Base):
    """Campaña Smartlead usada para altas nuevas (sync HubSpot → push). Historial vía `status`."""

    __tablename__ = "campaign_active"

    id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True),
        primary_key=True,
        default=uuid.uuid4,
    )
    id_campaign: Mapped[str] = mapped_column(String(64), nullable=False)
    status: Mapped[str] = mapped_column(
        String(16),
        nullable=False,
        server_default=text("'Inactive'"),
    )

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
        nullable=False,
    )

File: frontend/src/types/campaignActive.ts
export type CampaignActiveRow = {
  id: string;
  id_campaign: string;
  status: string;
  created_at: string;
  updated_at: string;
};

export type CampaignActiveGetResponse = {
  active: CampaignActiveRow | null;
  effective_id_campaign: string;
};

export type CampaignActiveSetResponse = CampaignActiveRow & {
  effective_id_campaign: string;
};
